### PR TITLE
Some automated code cleanup

### PR DESCRIPTION
**Pull request by cyian-1756**
# Category

This change is exactly one of the following (please change `[ ]` to `[x]`) to indicate which:
* [ ] a bug fix (Fix #...)
* [ ] a new Ripper
* [X] a refactoring
* [ ] a style change/fix

# Description

I ran IdeaC code analyst tool on ripme and let it auto fix a bunch of simple errors, as with any automated code cleanup this should be tested pretty heavily before being merged.

# Testing

Required verification:
* [ ] I've verified that there are no regressions in `mvn test` (there are no new failures or errors).
* [ ] I've verified that this change works as intended.
  * [ ] Downloads all relevant content.
  * [ ] Downloads content from multiple pages (as necessary or appropriate).
  * [ ] Saves content at reasonable file names (e.g. page titles or content IDs) to help easily browse downloaded content.
* [ ] I've verified that this change did not break existing functionality (especially in the Ripper I modified).

Optional but recommended:
* [ ] I've added a unit test to cover my change.

### CODE REVIEW

File: src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java [hunk 1]
```diff
@@ -142,7 +143,7 @@ public void rip() throws IOException {
     }
     private String fileNameFromURL(URL url) {
         String saveAs = url.toExternalForm();
-        if (saveAs.substring(saveAs.length() - 1) == "/") { saveAs = saveAs.substring(0,saveAs.length() - 1) ;}
+        if (Objects.equals(saveAs.substring(saveAs.length() - 1), "/")) { saveAs = saveAs.substring(0,saveAs.length() - 1) ;}
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java]: shouldn't `String#equals` (i.e. `a.equals(b)`) be applicable here?

File: src/main/java/com/rarchives/ripme/ripper/rippers/CfakeRipper.java [hunk 2]
```diff
@@ -59,7 +60,7 @@ public Document getNextPage(Document doc) throws IOException {
                 String nextPage = elem.attr("href");
                 // Some times this returns a empty string
                 // This for stops that
-                if (nextPage == "") {
+                if (Objects.equals(nextPage, "")) {
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/CfakeRipper.java]: same goes here. might as well fix the above comment as well.

File: src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java (hunk 1 above)
Comment (metaprime) [reply src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java]: (Same comment applies to all other usages of `Objects.equals`)

File: src/main/java/com/rarchives/ripme/ripper/rippers/EightmusesRipper.java [hunk 3]
```diff
@@ -3,11 +3,7 @@
 import java.io.IOException;
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/EightmusesRipper.java]: This "automated fix" depends on what your IDE "prefers", but I generally see `import`s canonicalized to not use wildcards (*), and I think I prefer that.

File: src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java (hunk 1 above)
Comment (cyian-1756) [reply src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java]: > shouldn't String#equals (i.e. a.equals(b)) be applicable here?

I've no idea. I don't know all that much about Java

File: src/main/java/com/rarchives/ripme/ripper/rippers/EightmusesRipper.java (hunk 3 above)
Comment (cyian-1756) [reply src/main/java/com/rarchives/ripme/ripper/rippers/EightmusesRipper.java]: I agree.

File: src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java (hunk 1 above)
Comment (metaprime) [reply src/main/java/com/rarchives/ripme/ripper/AbstractHTMLRipper.java]: They are both `String`s so I think we should do `<leftstring>.equals(<rightstring>)` in all these cases. (Which calls the `String#equals` method.)

**Pull request closed without merging**